logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/ObserveConversationInteractionAvailabilityUseCase.kt

```diff
@@ -66,6 +89,23 @@ class ObserveConversationInteractionAvailabilityUseCase internal constructor(
             })
         }
     }
+
+    private fun doesUserSupportConversationProtocol(
+        conversationDetails: ConversationDetails,
+        selfUser: SelfUser
+    ): Boolean {
+        val protocolInfo = conversationDetails.conversation.protocol
+        val acceptableProtocols = when (protocolInfo) {
+            is Conversation.ProtocolInfo.MLS -> setOf(SupportedProtocol.MLS)
+            // Messages in mixed conversations are sent through Proteus
+            is Conversation.ProtocolInfo.Mixed -> setOf(SupportedProtocol.PROTEUS)
+            Conversation.ProtocolInfo.Proteus -> setOf(SupportedProtocol.PROTEUS)
+        }
+        val isProtocolSupported = selfUser.supportedProtocols?.any { supported ->
+            acceptableProtocols.contains(supported)
+        } ?: false
```

Comment:
what the point of the `set` in `when`? Can be like:

```
        val protocolInfo = conversationDetails.conversation.protocol
        val isProtocolSupported = when (protocolInfo) {
            is Conversation.ProtocolInfo.MLS -> selfUser.supportedProtocols?.any { it == SupportedProtocol.MLS) }
            // Messages in mixed conversations are sent through Proteus
            is Conversation.ProtocolInfo.Mixed,
            Conversation.ProtocolInfo.Proteus -> selfUser.supportedProtocols?.any { it == SupportedProtocol.PROTEUS) }
        } ?: false
```

Just a suggestion, not a mandatory :)

Reply:
Oooh yeah.

Originally I added `setOf(MLS, PROTEUS)` for `Mixed`. Then I confirmed that `Mixed` requires `Proteus` only.

Indeed, it can be simplified.

Nice catch!